lib/remote/apilistener-configsync.cpp

```diff
@@ -462,3 +483,39 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Finished syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 }
+
+void ConfigSyncRule::WaitForObject(const Type::Ptr& ptype, const String& objName)
+{
+	std::unique_lock<std::mutex> lock(m_Mutex);
+	m_CV.wait(lock, [this, &ptype, &objName]{ return !IsObjectBlocked(ptype, objName); });
+}
+
+bool ConfigSyncRule::IsObjectBlocked(const Type::Ptr& ptype, const String& objName)
+{
+	auto type = m_Objects.find(ptype.get());
+	if (type == m_Objects.end()) {
+		return false;
+	}
+
+	return type->second.find(objName) != type->second.end();
+}
+
+void ConfigSyncRule::BlockObject(const Type::Ptr& ptype, const String& objName)
+{
+	std::unique_lock<std::mutex> lock(m_Mutex);
+	m_Objects[ptype.get()].emplace(objName);
+}
+
+void ConfigSyncRule::UnBlockNotify(const Type::Ptr& ptype, const String& objName, bool notifyAll)
+{
+	{
+		std::unique_lock<std::mutex> lock(m_Mutex);
+		m_Objects[ptype.get()].erase(objName);
+	}
+
+	if (notifyAll) {
+		m_CV.notify_all();
+	} else {
+		m_CV.notify_one();
+	}
```

Comment:
You don't seem to ever set `notifyAll = true`, that seems to be unintentional. Also, I was already wondering before why you used `notify_all()` in some places and `notify_all()` in others. I don't see a way (at least without quite some extra complexity) to ensure that `notify_one()` would wake the correct thread.